### ISSUE TITLE
Error in deserializing body of reply message for operation 'catalogProductRepositoryV1GetList'

### ISSUE THREAD

**Issue by resmivarghese** [magento/magento2#5949]
HI,
We are adding magento2 services in .Net and it works for few services and for few its failing. Belo is the detail.
Version : Magento 2.0.5
I have added service reference in C#.Net for the service catalogProductRepositoryV1, When running from SOAP UI I get the result. But when calling from .Net code it throws exception as below:
Service name : http://{baseurl}/default/soap/default?services=catalogProductRepositoryV1
Error in deserializing body of reply message for operation 'catalogProductRepositoryV1GetList'.
There is an error in XML document (2, 705).Input string was not in a correct format.
Below is the .Net Code:
`

var frameworkFilter = new product.FrameworkFilter()

```
{
field = "sku",
value = "24-MB01"
};
var frameworkFilters = new product.FrameworkFilter[]
{
frameworkFilter
};


var frameworkSortOrder = new product.FrameworkSortOrder()
            {
                field = "sku",
                direction = "asc"
            };
             var frameworkSortOrders = new product.FrameworkSortOrder[]
            {
                frameworkSortOrder  
            };
```

```
        var frameworkSearchFilterGroup = new product.FrameworkSearchFilterGroup()
        {
            filters = frameworkFilters
        };
        var frameworkSearchFilterGroups = new product.FrameworkSearchFilterGroup[]
    {
        frameworkSearchFilterGroup
    };
        var frameworkSearchCriteriaInterface = new product.FrameworkSearchCriteriaInterface()
        {

            filterGroups = frameworkSearchFilterGroups,
            sortOrders=frameworkSortOrders,
             pageSize = 10,
            pageSizeSpecified = true,
            currentPage = 1,
            currentPageSpecified = true
        };

        var catalogProductRepositoryV1GetListRequest = new product.CatalogProductRepositoryV1GetListRequest()
        {
            searchCriteria = frameworkSearchCriteriaInterface
        };
```

`product.CatalogProductRepositoryV1GetListResponse response = Magento2HeaderSettings.Product.catalogProductRepositoryV1GetList(catalogProductRepositoryV1GetListRequest);`

`
Config details:
binding="customBinding" bindingConfiguration="catalogProductRepositoryV1Binding"
contract="Mag2ProductServiceReference.catalogProductRepositoryV1PortType"
name="catalogProductRepositoryV1Port" />
`
I have tried all the possibilities but nothing worked. But I am getting correct response when trying with customerCustomerRepositoryV1.
Any help is appreciated.
Thanks

**Comment by manhao-chen** [comment]
it may be related to https://github.com/magento/magento2/issues/3605

**Comment by andimov** [comment]
@resmivarghese
Thank you for reporting!
To allow us to research the issue, please, format this issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines).

**Comment by resmivarghese** [comment]
**Title**
Error in deserializing body of reply message for operation 'salesOrderRepositoryV1Get'.

**Issue Description**

We have a .Net application which needs to be communicated to Magento2 webservices.
Iam getting exception while calling the webservice salesOrderRepositoryV1Get from .Net. 
In this ticket I have first reported for catalogProductRepositoryV1GetList which was in version Magento 2.0.5.After reading the related forums, we upgraded to magento 2.1.0 and the service catalogProductRepositoryV1GetList started working.So we continued our development .But now we are struck with next service Sales order services.

MagentoVersion : 2.1.0
Visual Studio 2013.

**Steps to reproduce**

Create a .Net application .Add service reference of ~/soap/en?services=salesOrderRepositoryV1. Below is the complete code.

```
 class Magento2HeaderSettings
    {
        const string MAGENTO_SOAP_ACCESS_TOKEN = "22pcf6wtypa8m76rgvy7061fbdk3ltma";
        private static BasicHttpBinding GetBinding()
        {
            var bindingElement = new TextMessageEncodingBindingElement();
            bindingElement.ReaderQuotas.MaxArrayLength = 65536;

            var binding = new BasicHttpBinding();
            binding.ReaderQuotas.MaxArrayLength = 2147483647;
            binding.ReaderQuotas.MaxStringContentLength = 2147483647;
            binding.ReaderQuotas.MaxDepth = 2147483647;
            binding.ReaderQuotas.MaxBytesPerRead = 2147483647;
            binding.ReaderQuotas.MaxNameTableCharCount = 2147483647;
           return binding;
           //return new BasicHttpBinding();
        }

        private static void AddAuthorizationHeader(IClientChannel clientChannel)
        {
            HttpRequestMessageProperty hrmp = new HttpRequestMessageProperty();
            hrmp.Headers.Add("Authorization", "Bearer " + MAGENTO_SOAP_ACCESS_TOKEN);

            OperationContextScope contextScope = new OperationContextScope(clientChannel);
            OperationContext.Current.OutgoingMessageProperties[HttpRequestMessageProperty.Name] = hrmp;
        }

        public static salesOrderRepositoryV1PortTypeClient SalesOrder
        {
            get
            {
                var client = new salesOrderRepositoryV1PortTypeClient();
                client.Endpoint.Binding = GetBinding();
                AddAuthorizationHeader(client.InnerChannel);
                return client;
            }
        }


    }
```

```
class MagentoService
{
public static Magento2OrderServiceReference.SalesOrderRepositoryV1GetResponse GetSalesOrderById(int id)
    {
try
{
    **SalesOrderRepositoryV1GetResponse salesOrder = Magento2HeaderSettings.SalesOrder.salesOrderRepositoryV1Get(new Citrusstv.ESB.Services.Magento2OrderServiceReference.SalesOrderRepositoryV1GetRequest { id = int.Parse(request.OrderId) });**
}
catch (Exception ex)
 {}
    }
}
```

**Config file:**

```
<client>
<endpoint address="http://www.test1.net/soap/en?services=salesOrderRepositoryV1"
        binding="customBinding" bindingConfiguration="salesOrderRepositoryV1Binding"
        contract="Magento2OrderServiceReference.salesOrderRepositoryV1PortType"
        name="salesOrderRepositoryV1Port" />
</client>       
<customBinding> 
    <binding name="salesOrderRepositoryV1Binding">
      <textMessageEncoding maxReadPoolSize="64" maxWritePoolSize="16"
      messageVersion="Soap12" writeEncoding="utf-8">
        <readerQuotas maxDepth="2147483647" maxStringContentLength="2147483647"
                      maxArrayLength="2147483647" maxBytesPerRead="2147483647"
                      maxNameTableCharCount="2147483647" />
      </textMessageEncoding>
      <httpTransport />
    </binding>
</customBinding>
```

Exception throws when we call the operation salesOrderRepositoryV1Get.

**Inner Exception** **_{"Input string was not in a correct format."}_**
**Message** **_There is an error in XML document (2, 485)._**
**[System.ServiceModel.CommunicationException]** **_"Error in deserializing body of reply message for operation 'salesOrderRepositoryV1Get'._** 

I have checked multiple forums and it says to increas teh readerquotes values.But its not helping me.
Appreciating the support and help .

Thanks,

**Comment by resmivarghese** [comment]
We are facing same deserializing issue in Shoppingcart services([`quoteCartRepositoryV1`]() also.

For the operation `quoteCartRepositoryV1Save` ,along with Deserializing issue another exception also throws when we add  service in .Net . 

Below is the error details,

> For save Additional information: The method 'quoteCartRepositoryV1Save' in type 'Magento2API.QuoteCartRepositoryServiceReference.quoteCartRepositoryV1PortType' is marked IsOneWay=true and declares one or more FaultContractAttributes. One-way methods cannot declare FaultContractAttributes. To fix it, change IsOneWay to false or remove the FaultContractAttributes.

Below is code added 

```
quoteCartRepositoryV1PortTypeClient client = new quoteCartRepositoryV1PortTypeClient();
                client.Endpoint.Binding = new BasicHttpBinding();
                QuoteCartRepositoryV1GetRequest request = new QuoteCartRepositoryV1GetRequest();
                request.cartId = 431843;
                var response = client.quoteCartRepositoryV1Get(request);
```

Looking forward for an update on these issues.

**Comment by veloraven** [comment]
@resmivarghese is this issue still actual?
If it is please format it according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.
If it is just a question and not an issue please refer to the [Community Forums](https://community.magento.com) or the [Magento Stack Exchange](http://magento.stackexchange.com/) as GitHub issue tracker is intended for technical issues only.

**Comment by veloraven** [comment]
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
If this issue still reproducible please feel free to create the new one: format new issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result and specify Magento version.

**Comment by ilopez** [comment]
I have the same problem, but this is how I discovered it and resolved it on my end.

**Problem**
Unable to deserialze SOAP Message.
**My Cause**
Unable to parse string to int
**Root Cause**
String XML tag is null instead of 0.

**Observations**
XML String Error offset (Line, Char)

**Steps to Reproduce**
1. Create an order
2. Simulate Payment with a broken implementation for Notes
3. Observe new order comment with "is Customer Notified" is XML null instead of 0 / 1.

`<isCustomerNotified/>`

**Workaround**
Per #6854 I modified my previously saved WSDL and found the offending XSD Definition. isCustomerNotified from type int to type string.

**Proposed Fix**
Somewhere between the database record, and the SOAP response, a null value should not propagate through especially if the WSDL forbids it.

**Additional Notes**
Using SoapUI I was able to determine the exact offending line much better than Visual Studio Generated Code. It supplied an error message with:

`line 218: Invalid decimal value: expected at least one digit`

For Soap UI, you do have the opportunity to specify the Bearer Authorization Token, and generate an authenticated request. Right click on the response XML window and select "Validate" or [ALT+V].
